File: src/utils/battleUtils.js
export const calculateDamage = (attacker, defender) => {
  const baseDamage = attacker.meta.stats.Attack - defender.meta.stats.Defense;
  const criticalHit = Math.random() < attacker.meta.stats['Critical Chance'] ? 1.5 : 1;
  const damage = Math.max(0, baseDamage * criticalHit);
  return damage;
};

export const determineTurnOrder = (ordinooki1, ordinooki2) => {
  if (ordinooki1.meta.stats.Speed > ordinooki2.meta.stats.Speed) {
    return [ordinooki1, ordinooki2];
  } else if (ordinooki1.meta.stats.Speed < ordinooki2.meta.stats.Speed) {
    return [ordinooki2, ordinooki1];
  } else {
    return Math.random() < 0.5 ? [ordinooki1, ordinooki2] : [ordinooki2, ordinooki1];
  }
};

File: backend/utils/battle_utils.py
# utils/battle_utils.py

def calculate_damage(attacker, defender):
    """
    Calculate damage based on attacker and defender stats.
    Implement your logic here.
    """
    # Example logic (replace with actual game mechanics)
    base_damage = 10
    return base_damage

def determine_turn_order(player1, player2):
    """
    Determine which player attacks first based on speed stats or other criteria.
    Returns a list [first_player, second_player]
    """
    # Example logic: Player with higher speed goes first
    speed1 = player1['ordinooki'].get('meta', {}).get('stats', {}).get('Speed', 10)
    speed2 = player2['ordinooki'].get('meta', {}).get('stats', {}).get('Speed', 10)
    if speed1 >= speed2:
        return [player1, player2]
    else:
        return [player2, player1]
